posthog/temporal/compare_recording_metadata/compare_recording_metadata_workflow.py

```diff
@@ -0,0 +1,271 @@
+            sum(console_log_count) as console_log_count,
+            sum(console_warn_count) as console_warn_count,
+            sum(console_error_count) as console_error_count,
+            sum(size) as size,
+            sum(message_count) as message_count,
+            sum(event_count) as event_count,
+            argMinMerge(snapshot_source) as snapshot_source,
+            argMinMerge(snapshot_library) as snapshot_library
+            {block_fields}
+        FROM
+            {table}
+        GROUP BY
+            session_id
+        HAVING
+            min_first_timestamp >= toDateTime(%(started_after)s)
+            AND min_first_timestamp <= toDateTime(%(started_before)s)
+    """
+
+    # Add block-related fields only for v2 table
+    block_fields = (
+        """
+        ,groupArrayArray(block_first_timestamps) as block_first_timestamps,
+        groupArrayArray(block_last_timestamps) as block_last_timestamps,
+        groupArrayArray(block_urls) as block_urls
+    """
+        if "_v2_" in table_name
+        else ""
+    )
+
+    return sync_execute(
+        query.format(table=table_name, block_fields=block_fields),
+        {
+            "started_after": started_after.strftime("%Y-%m-%d %H:%M:%S"),
+            "started_before": started_before.strftime("%Y-%m-%d %H:%M:%S"),
+        },
+    )
+
+
+FIELD_NAMES = [
+    "distinct_id",
+    "min_first_timestamp",
+    "max_last_timestamp",
+    "first_url",
+    "all_urls",
+    "click_count",
+    "keypress_count",
+    "mouse_activity_count",
+    "active_milliseconds",
+    "console_log_count",
+    "console_warn_count",
+    "console_error_count",
+    "size",
+    "message_count",
+    "event_count",
+    "snapshot_source",
+    "snapshot_library",
+]
+
+
+@dataclasses.dataclass(frozen=True)
+class CompareRecordingMetadataActivityInputs:
+    """Inputs for the `compare_recording_metadata_activity`."""
+
+    started_after: str = dataclasses.field()
+    started_before: str = dataclasses.field()
+
+    @property
+    def properties_to_log(self) -> dict[str, typing.Any]:
+        return {
+            "started_after": self.started_after,
+            "started_before": self.started_before,
+        }
+
+
+@temporalio.activity.defn
+async def compare_recording_metadata_activity(inputs: CompareRecordingMetadataActivityInputs) -> None:
+    """Compare session recording metadata between storage backends."""
+    async with Heartbeater():
+        logger = get_internal_logger()
+        started_after = dt.datetime.fromisoformat(inputs.started_after)
+        started_before = dt.datetime.fromisoformat(inputs.started_before)
+
+        results_v1 = get_session_replay_events(
+            "session_replay_events",
+            started_after,
+            started_before,
+        )
+        results_v2 = get_session_replay_events(
+            "session_replay_events_v2_test",
+            started_after,
+            started_before,
+        )
```

Comment:
These calls look like blocking I/O that will block the entire worker. I would recommend either sending them to a thread with `asyncio.to_thread`, making the whole activity synchronous, or re-writing `get_session_replay_events` to be non-blocking.

Reply:
Moved the queries to threads, I think rewriting to use an async client would be too much effort for this.